### ISSUE TITLE
Incomplete OGC:WMS Download Link for Catalog

### ISSUE THREAD

**Issue by afabiani** [GeoNode/geonode#2720]
GeoNode download_links for catalog, does not include the fully qualified layer name and therefore it is not possible for catalog clients to recognize the geospatial layer.

When generating the OGC:WMS download link, GeoNode should append the fully qualified layer name also.

**Comment by tomkralidis** [comment]
@afabiani we are not spelling out full GetMap requests by design.  The idea is for downstream clients to grab the URL and layer name and bind.

**Comment by simboss** [comment]
@tomkralidis I agree with your point, I'll talk to Alessio about the PR.

The one problem I see is that the way the information is conveyed from the CSW uses a specific GeoServer features that is virtual services. The WMS link uses the geonode workspace to qualify the service base link and the name is not workspace qualified. 

This is _ok_ since GeoNode stores data in a single workspace but can get confusing if you had adta in other workspace. 

Anywyay, this is not a big deal for the moment.

**Comment by afabiani** [comment]
Main issue I see is that GeoNode CSW does not declare the real Layer name at all except in few download links which can be different from a dataset to another. The way I modified the behavior is actually similar to the GeoNetwork approach.

**Comment by afabiani** [comment]
@simboss the resource name provided by the fix, does not currently rely on the fixed geonode workspace, but it declares the fully qualified name of the resource as stored in the GeoNode backend.

**Comment by afabiani** [comment]
@tomkralidis @simboss @pjdufour I would like to come back to this issue.

Currently I have taken a look at the GeoNode and PyCSW integration and trying to figure out alternatives to this one here

https://github.com/GeoNode/geonode/pull/2721

bu honestly I could not find a suitable solution for this issue.

I also have seen this topic here

https://github.com/geopython/pycsw/wiki/Geonode-notes

which basically propose the same thing, i.e. specify the full WMS link on the OGC:WMS reference.

@tomkralidis can you please specify your concerns about this? And can you also propose alternatives?

**Comment by tomkralidis** [comment]
@afabiani thanks for the info.  What is the core workflow here?  For a client to discover a metadata record and bind to it? Using GeoNode's ISO output (which provides the given OWS URLs as well as Layer/FeatureType/Coverage name) is the clear and definitive way to do this, allowing the client to assemble requests accordingly.

**Comment by afabiani** [comment]
Hi @tomkralidis thanks for replying.
Currently there's no Layer/FeatureType/Coverage name in GeoNode, this is the issue.

**Comment by simboss** [comment]
@tomkralidis I agree that using ISO is better suited, especially when coming from the geospatial world, but there is a number of fields where DC is the standard and I think it would be valuable for GeoNode if we could expose services ' endpoints through DC as well.

For example here in IT there is a number of OpenData initiatives that use DC as the way to share metadata and we would really want to push GeoNode as way to publish their data but we need to be able to have services endpoints as well.

What do you think?
